PrestaShop/classic-theme
# Implement picture tags

**This should go to 2.1.x and develop.**

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | This implements picture tags where possible and adds webp+avif sources, if available. Adds sources info to JS attributes to use it in JS. Change image titles and alts properly.
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Sponsor company   | 
| How to test?      | See below

### What to ignore during testing
- Picture tags are implemented only on images where the sources were provided. Exceptions are for example:
  - Manufacturer logo on product page, manufacturer images in manufacturers page, supplier listing. This will be fixed by introducing a manufacturer presenter.
- It does not work correctly for category images and store images, they may contain JPG links inside the webp sources. It will be fixed by https://github.com/PrestaShop/PrestaShop/pull/31310. However, the code is there and prepared. 👍

### How to test
- Add some images to a product and fill legends to those images.
- Go to product page and inspect product main image.
- See that there is a picture tag now with the img tag inside.
- See that if you click thumbnails, the ALT and TITLE attributes of the IMG tag are now changing properly.
- See that if you enable the new image system in BO and enable JPG and WEBP, you will see the webp appear as a source tag inside the picture tag.
- See that if you click thumbnails now, it changes also the source tags.
- Also go to stores list and try to inspect an image of some store. You will see that it's now a picture tag with img and sources inside. The webp/png/avif sources may contain jpg links, depending on if https://github.com/PrestaShop/PrestaShop/pull/31310 is merged or not.

### Video
https://user-images.githubusercontent.com/6097524/217795030-07ec30b8-492a-4503-a00e-76aac00381f0.mp4
